plugin/src/main/java/io/jenkins/plugins/analysis/core/restapi/ToolApi.java

```diff
@@ -28,12 +33,17 @@
      *         the URL to the latest results
      * @param size
      *         the number of warnings
+     * @param sizePerSeverity
+     *         the number of warnings, grouped by severity. Assumption is that
+     *         this is not null, and that it is properly populated with counts
+     *         that when summed, total the "size" parameter.
      */
-    public ToolApi(final String id, final String name, final String latestUrl, final int size) {
+    public ToolApi(final String id, final String name, final String latestUrl, final int size, final Map<Severity, Integer> sizePerSeverity) {
```

Comment:
Do not delete the old constructor, rather mark it as deprecated and call the new one with an empty map so that other consumers of this API still need no changes. (See build failure due to revapi.)

Reply:
Thanks for the feedback. I have made the change in 4566a73e2088af70276a83271107f5f3c31775ad.